<launch>
  <!-- rqt_reconfigure -->
  <arg name="obstacleAvoidance/rqt_reconfigure" default="true"/>
  <!-- 探索種類 -->
  <arg name="obstacleAvoidance/SEARCH_ONLY_ANGLE" default="true"/>
  <!-- ロボットパラメータ -->
  <arg name="obstacleAvoidance/distance_th" default="2.0"/>
  <arg name="obstacleAvoidance/WheelD" default="0.3125"/>
  <arg name="obstacleAvoidance/angleMin" default="45"/>
  <arg name="obstacleAvoidance/angleMax" default="135"/>
  <arg name="obstacleAvoidance/angleDiv" default="1.0"/>
  <arg name="obstacleAvoidance/maxSpeed" default="0.30"/>
  <arg name="obstacleAvoidance/minSpeed" default="0.10"/>
  <arg name="obstacleAvoidance/defaultSpeed" default="0.3"/>
  <!-- ゴール位置 -->
  <arg name="obstacleAvoidance/goalX" default="0.0"/>
  <arg name="obstacleAvoidance/goalY" default="6.0"/>
  <!-- vfh -->
  <!-- vfh - k -->
  <arg name="obstacleAvoidance/marginRadius" default="1.0"/>
  <arg name="obstacleAvoidance/Kcp" default="1.0"/>
  <arg name="obstacleAvoidance/Kg" default="1.0"/>
  <arg name="obstacleAvoidance/KcurAngle" default="0.25"/>
  <arg name="obstacleAvoidance/KprevAngle" default="0.25"/>
  <!-- vfh - eta -->
  <arg name="obstacleAvoidance/EtaCp" default="1"/>
  <arg name="obstacleAvoidance/EtaG" default="1"/>
  <arg name="obstacleAvoidance/EtaCurAngle" default="1"/>
  <arg name="obstacleAvoidance/EtaPrevAngle" default="1"/>
  <!-- safe range -->
  <arg name="obstacleAvoidance/safeRange" default="1.2"/>
  <!-- 探査範囲 -->
  <arg name="obstacleAvoidance/searchRange_vel" default="0.2"/>
  <arg name="obstacleAvoidance/searchDiv_vel" default="0.01"/>
  <!-- debug -->
  <arg name="obstacleAvoidance/debugType" default="1"/>
  <arg name="robot_name" default="robot1"/>

  <arg name="obstacleAvoidance/CONTROLLER/GAIN/P" default="0.20"/>
  <arg name="obstacleAvoidance/POTENTIAL/F1" default="0"/>
  <arg name="obstacleAvoidance/POTENTIAL/F2" default="0"/>
  <arg name="obstacleAvoidance/MOVE_MEAN/WINDOW_NUM" default="10"/>

  <group ns="$(arg robot_name)">
  <node pkg="local_navigation" name="local_navigation_cp_vfh" type="local_navigation_cp_vfh" output="screen" respawn="true">
    <!-- rqt_reconfigure -->
    <param name="obstacleAvoidance/rqt_reconfigure" value="$(arg obstacleAvoidance/rqt_reconfigure)"/>
    <!-- 探索種類 -->
    <param name="obstacleAvoidance/SEARCH_ONLY_ANGLE" value="$(arg obstacleAvoidance/SEARCH_ONLY_ANGLE)"/>
    <!-- ロボットパラメータ -->
    <param name="obstacleAvoidance/WheelD" value="$(arg obstacleAvoidance/WheelD)"/>
    <param name="obstacleAvoidance/angleMin" value="$(arg obstacleAvoidance/angleMin)"/>
    <param name="obstacleAvoidance/angleMax" value="$(arg obstacleAvoidance/angleMax)"/>
    <param name="obstacleAvoidance/angleDiv" value="$(arg obstacleAvoidance/angleDiv)"/>
    <param name="obstacleAvoidance/maxSpeed" value="$(arg obstacleAvoidance/maxSpeed)"/>
    <param name="obstacleAvoidance/minSpeed" value="$(arg obstacleAvoidance/minSpeed)"/>
    <param name="obstacleAvoidance/defaultSpeed" value="$(arg obstacleAvoidance/defaultSpeed)"/>
    <!-- ゴール位置 -->
    <param name="obstacleAvoidance/goalX" value="$(arg obstacleAvoidance/goalX)"/>
    <param name="obstacleAvoidance/goalY" value="$(arg obstacleAvoidance/goalY)"/>
    <!-- vfh -->
    <param name="obstacleAvoidance/distance_th" value="$(arg obstacleAvoidance/distance_th)"/>
    <!-- vfh - k -->
    <param name="obstacleAvoidance/marginRadius" value="$(arg obstacleAvoidance/marginRadius)"/>
    <param name="obstacleAvoidance/Kcp" value="$(arg obstacleAvoidance/Kcp)"/>
    <param name="obstacleAvoidance/Kg" value="$(arg obstacleAvoidance/Kg)"/>
    <param name="obstacleAvoidance/KcurAngle" value="$(arg obstacleAvoidance/KcurAngle)"/>
    <param name="obstacleAvoidance/KprevAngle" value="$(arg obstacleAvoidance/KprevAngle)"/>
    <!-- vfh - eta -->
    <param name="obstacleAvoidance/EtaCp" value="$(arg obstacleAvoidance/EtaCp)"/>
    <param name="obstacleAvoidance/EtaG" value="$(arg obstacleAvoidance/EtaG)"/>
    <param name="obstacleAvoidance/EtaCurAngle" value="$(arg obstacleAvoidance/EtaCurAngle)"/>
    <param name="obstacleAvoidance/EtaPrevAngle" value="$(arg obstacleAvoidance/EtaPrevAngle)"/>
    <!-- safe range -->
    <param name="obstacleAvoidance/safeRange" value="$(arg obstacleAvoidance/safeRange)"/>
    <!-- 探査範囲 -->
    <param name="obstacleAvoidance/searchRange_vel" value="$(arg obstacleAvoidance/searchRange_vel)"/>
    <param name="obstacleAvoidance/searchDiv_vel" value="$(arg obstacleAvoidance/searchDiv_vel)"/>
    <!-- debug -->
    <param name="obstacleAvoidance/debugType" value="$(arg obstacleAvoidance/debugType)"/>
    <!-- 2023年度　追加　石川 -->
    <param name="obstacleAvoidance/CONTROLLER/GAIN/P" value="$(arg obstacleAvoidance/CONTROLLER/GAIN/P)"/>
    <param name="obstacleAvoidance/POTENTIAL/F1" value="$(arg obstacleAvoidance/POTENTIAL/F1)"/>
    <param name="obstacleAvoidance/POTENTIAL/F2" value="$(arg obstacleAvoidance/POTENTIAL/F2)"/>
  </node>
  </group>
</launch>
